packages/common/src/store/buy-crypto/sagas.ts

```
@@ -0,0 +1,497 @@
+      {
+        swapRequest: {
+          quoteResponse,
+          userPublicKey: wallet.publicKey.toBase58(),
+          destinationTokenAccount: userbank.toBase58(),
+          useSharedAccounts: true
+        }
+      }
+    )
+
+    // Close the temporary wSOL account
+    const closeWSOLInstruction = createCloseAccountInstruction(
+      walletSolTokenAccount, //  account to close
+      feePayer, // fee destination
+      wallet.publicKey //  owner
+    )
+
+    const instructions = [
+      memoInstruction,
+      createWSOLInstruction,
+      transferWSOLInstruction,
+      syncNativeInstruction,
+      parseJupiterInstruction(swapInstruction),
+      closeWSOLInstruction
+    ]
+    const { transaction, addressLookupTableAccounts } = yield* call(
+      createVersionedTransaction,
+      audiusBackendInstance,
+      {
+        instructions,
+        lookupTableAddresses: addressLookupTableAddresses,
+        feePayer
+      }
+    )
+    transaction.sign([wallet])
+
+    const { error } = yield* call(
+      relayVersionedTransaction,
+      audiusBackendInstance,
+      {
+        transaction,
+        addressLookupTableAccounts,
+        skipPreflight: true
+      }
+    )
+
+    if (error) {
+      throw new BuyCryptoError(BuyCryptoErrorCode.SWAP_ERROR, error)
+    }
+
+    // Record success
+    if (mint === 'audio') {
+      yield* call(
+        track,
+        make({
+          eventName: Name.BUY_AUDIO_SUCCESS,
+          provider,
+          requestedAudio: amount,
+          surplusAudio:
+            amount - Number(quoteResponse.outAmount) / outputTokenLamports,
+          actualAudio: Number(quoteResponse.outAmount) / outputTokenLamports
+        })
+      )
+    } else if (mint === 'usdc') {
+      yield* call(
+        track,
+        make({
+          eventName: Name.BUY_USDC_SUCCESS,
+          provider,
+          requestedAmount: amount
+        })
+      )
+    }
+    yield* put(buyCryptoSucceeded())
```

maybe else { throw new Error(`Invalid mint ${mint}`) or something

---

I don't think we get this far with invalid mint fwiw. Might make sense though to have up front guard